roboplot/core/stepper_control.py

```diff
@@ -49,16 +73,89 @@ def forwards(self) -> bool:
     def forwards(self, value: bool) -> None:
         self._motor.clockwise = value != self._invert_axis
 
+    @property
+    def is_homed(self):
+        return self._is_homed
+
+    def home(self) -> None:
+        """
+        Home the axis by driving into the limit switch and setting the current_location upon reaching it.
+
+        The home_position argument to Axis.__init__ controls the direction in which to home as well as the value set
+        upon reaching it.
+        """
+        self.forwards = self._home_position.forwards
```

Comment:
Add comments so I know what each block of code is doing - e.g.  # Check that no limit switch is currently pressed.